src/routes/v3/payouts.rs

```diff
@@ -0,0 +1,732 @@
+                    result.user_id
+                )
+                .execute(&mut *transaction)
+                .await?;
+
+                crate::database::models::user_item::User::clear_caches(
+                    &[(crate::database::models::UserId(result.user_id), None)],
+                    &redis,
+                )
+                .await?;
+
+                sqlx::query!(
+                    "
+                    UPDATE payouts
+                    SET status = $1
+                    WHERE platform_id = $2
+                    ",
+                    if &*webhook.event == "REWARDS.CANCELED" {
+                        PayoutStatus::Cancelled
+                    } else {
+                        PayoutStatus::Failed
+                    }
+                    .as_str(),
+                    webhook.payload.resource.id
+                )
+                .execute(&mut *transaction)
+                .await?;
+
+                transaction.commit().await?;
+            }
+        }
+        "PAYMENT.PAYOUTS-ITEM.SUCCEEDED" => {
+            let mut transaction = pool.begin().await?;
+            sqlx::query!(
+                "
+                UPDATE payouts
+                SET status = $1
+                WHERE platform_id = $2
+                ",
+                PayoutStatus::Success.as_str(),
+                webhook.payload.resource.id
+            )
+            .execute(&mut *transaction)
+            .await?;
+            transaction.commit().await?;
+        }
+        _ => {}
+    }
+
+    Ok(HttpResponse::NoContent().finish())
+}
+
+#[get("")]
+pub async fn user_payouts(
+    req: HttpRequest,
+    pool: web::Data<PgPool>,
+    redis: web::Data<RedisPool>,
+    session_queue: web::Data<AuthQueue>,
+) -> Result<HttpResponse, ApiError> {
+    let user = get_user_from_headers(
+        &req,
+        &**pool,
+        &redis,
+        &session_queue,
+        Some(&[Scopes::PAYOUTS_READ]),
+    )
+    .await?
+    .1;
+
+    let payout_ids =
+        crate::database::models::payout_item::Payout::get_all_for_user(user.id.into(), &**pool)
+            .await?;
+    let payouts =
+        crate::database::models::payout_item::Payout::get_many(&payout_ids, &**pool).await?;
+
+    Ok(HttpResponse::Ok().json(
+        payouts
+            .into_iter()
+            .map(crate::models::payouts::Payout::from)
+            .collect::<Vec<_>>(),
+    ))
+}
+
+#[derive(Deserialize)]
+pub struct Withdrawal {
+    #[serde(with = "rust_decimal::serde::float")]
+    amount: Decimal,
+    method: PayoutMethodType,
+    method_id: String,
+}
+
+#[post("")]
+pub async fn create_payout(
+    req: HttpRequest,
+    pool: web::Data<PgPool>,
+    redis: web::Data<RedisPool>,
+    body: web::Json<Withdrawal>,
+    session_queue: web::Data<AuthQueue>,
+    payouts_queue: web::Data<PayoutsQueue>,
+) -> Result<HttpResponse, ApiError> {
+    let (scopes, user) =
+        get_user_record_from_bearer_token(&req, None, &**pool, &redis, &session_queue)
```

Comment:
Is there a reason this uses a separate get user to other routes ? (get_user_from_headers or w/e)

Reply:
yes it needs the raw paypal_id which aren't exposed in the public func